sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -1260,6 +1263,42 @@ bool _pi_queue::isBatchingAllowed() {
   return (this->QueueBatchSize > 0 && ((ZeSerialize & ZeSerializeBlock) == 0));
 }
 
+pi_result _pi_queue::getOrCreateCopyCommandQueue(
+    int Index, ze_command_queue_handle_t &ZeCopyCommandQueue) {
+  ZeCopyCommandQueue = nullptr;
+
+  // Make sure 'Index' is within limits
+  PI_ASSERT((Index >= 0) && (Index < (int)(ZeCopyCommandQueues.size())),
+            PI_INVALID_VALUE);
+
+  // Return the Ze copy command queue, if already available
+  if (ZeCopyCommandQueues[Index]) {
+    ZeCopyCommandQueue = ZeCopyCommandQueues[Index];
+    return PI_SUCCESS;
+  }
+
+  // Ze copy command queue is not available at 'Index'. So we create it below.
+  ZeStruct<ze_command_queue_desc_t> ZeCommandQueueDesc;
+  ZeCommandQueueDesc.ordinal = (Index == 0) ? Device->ZeMainCopyQueueGroupIndex
+                                            : Device->ZeLinkCopyQueueGroupIndex;
+  // There are two copy queues: main copy queues and link copy queues.
+  // ZeCommandQueueDesc.index is the index into the list of main (or link)
+  // copy queues.
+  // (Index == 0) means we are using the main copy queue.
+  // Otherwise, we use one of the link copy queues.
+  ZeCommandQueueDesc.index =
+      (Index == 0) ? 0 : Index - Device->hasMainCopyEngine();
```

Comment:
```suggestion
  ZeCommandQueueDesc.index = Index;
```
Why do you need to subtract anything? (even if needed you should not subtract a boolean)

Reply:
ZeCommandQueueDesc.index is the index into the list of Link Copy Engines (0-7). 'Index' is the index into the list of ALL copy engines (including main copy engine, if present) and it goes from 0-8 (or 0-7, if main copy engine is not available). So, we need to subtract 1 from Index (if main copy engine is present) to get ZeCommandQueueDesc.index for link copy engines.

Reply:
Is it really possible that there are "link" engines, but no "main" engine? If not, then ther will always be either 1 (0 index) engine or multiple (0 index being main, and >0 being link). Anyways, please do not treat a boolean "true" value as "1", as it is only "non-0" by standard.

Reply:
I replaced this with a hard '-1' for link copy indices and modified the comment. Thanks

Reply:
Thanks